dags/analyzer_helper/tests/unit/test_unit_discord_transform_raw_data.py

```diff
@@ -0,0 +1,175 @@
+            "role_mentions": ["role1", "role2"],
+            "reactions": ["user1, user2, :laugh:"],
+            "replied_user": None,
+            "createdDate": datetime(2024, 6, 10),
+            "messageId": "56789",
+            "channelId": "ghi",
+            "channelName": "💬・general-chat",
+            "threadId": "def",
+            "threadName": "thread-def",
+            "isGeneratedByWebhook": True,
+        }
+        interaction = self.transformer.create_interaction(
+            data=data,
+            name="reaction",
+            author="user123",
+            engaged_users=["user456"],
+            type="emitter",
+        )
+        self.assertEqual(
+            interaction,
+            {
+                "author_id": "user123",
+                "date": datetime(2024, 6, 10),
+                "source_id": "56789",
+                "metadata": {
+                    "thread_id": "def",
+                    "channel_id": "ghi",
+                    "bot_activity": True,
+                },
+                "actions": [],
+                "interactions": [
+                    {
+                        "name": "reaction",
+                        "users_engaged_id": ["user456"],
+                        "type": "emitter",
+                    }
+                ],
+            },
+        )
+
+    def test_create_transformed_item_valid_data(self):
+        data = {
+            "_id": ObjectId("649fc4dfb65f6981303e32ef"),
+            "type": 0,
+            "author": "user789",
+            "content": "sample message",
+            "user_mentions": ["user123"],
+            "role_mentions": ["role1", "role2"],
+            "reactions": ["user1, user2, :laugh:"],
+            "replied_user": None,
+            "createdDate": datetime(2024, 6, 11),
+            "messageId": "90123",
+            "channelId": "mno",
+            "channelName": "💬・general-chat",
+            "threadId": "jkl",
+            "threadName": "thread-jkl",
+            "isGeneratedByWebhook": False,
+        }
+        interaction = self.transformer.create_interaction(
+            data=data,
+            name="mention",
+            author="user789",
+            engaged_users=["user123"],
+            type="emitter",
+        )
+        interactions = interaction["interactions"]
+
+        transformed_item = self.transformer.create_transformed_item(
+            data=data, period=datetime(2024, 6, 11), interactions=[interaction]
+        )
+        expected_result = {
+            "author_id": "user789",
+            "date": datetime(2024, 6, 11),
+            "source_id": "90123",
+            "metadata": {
+                "thread_id": "jkl",
+                "channel_id": "mno",
+                "bot_activity": True,
+            },
+            "actions": [
+                {
+                    "name": "message",
+                    "type": "emitter",
+                }
+            ],
+            "interactions": interactions,
```

Comment:
Please do not use the received interactions within expected results. The interactions should be written manually so we could test it properly.